core/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java

```diff
@@ -312,6 +306,13 @@ void run() {
                     maybeFinish();
                 });
             }
+            if (sync) { // this is potentially slow so we do that first since it could safe us time on the waiting on refresh
```

Comment:
This comment is out of date now I think. The only advantage of waiting for refresh second is that if you have to wait for refresh and the refresh has already come through then the listener will be called immediately, with only a volatile read and no messing about with lists and queues and stuff.

It is probably less likely to matter now because fsync will be async in high throughput situations. So just nuke the comment and move on, I think.